packages/api/src/external/carequality/document/process-document-query-results.ts

```diff
@@ -82,86 +122,60 @@ export async function processDocumentQueryResults({
   }
 }
 
-function combineDocRefs(documentQueryResults: DocumentQueryResult[]): DocumentReference[] {
-  return documentQueryResults.reduce((acc: DocumentReference[], curr) => {
-    const documentReferences = curr.data.documentReference ?? [];
-
-    return [...acc, ...documentReferences];
-  }, []);
-}
-
-const getNonExistentDocRefs = async (
-  documents: DocumentReference[],
-  patientId: string,
-  cxId: string
-): Promise<DocumentReference[]> => {
-  const [{ existingDocRefs, nonExistingDocRefs }, FHIRDocRefs] = await Promise.all([
-    filterOutExistingDocRefsS3(documents, patientId, cxId),
-    getDocRefsFromFHIR(cxId, patientId),
-  ]);
-
-  const fhirDocRefIds = new Set(FHIRDocRefs.map(d => d.id));
-
-  const foundOnStorageButNotOnFHIR = existingDocRefs.filter(f => !fhirDocRefIds.has(f.docUniqueId));
-
-  const docsToDownload = nonExistingDocRefs.concat(foundOnStorageButNotOnFHIR);
-
-  return docsToDownload;
-};
-
-type ExistentialDocRefs = {
-  existingDocRefs: DocumentReference[];
-  nonExistingDocRefs: DocumentReference[];
-};
-
-const filterOutExistingDocRefsS3 = async (
-  documents: DocumentReference[],
-  patientId: string,
-  cxId: string
-): Promise<ExistentialDocRefs> => {
-  const docIdWithExist = await Promise.allSettled(
-    documents.map(async (doc): Promise<{ docId: string; exists: boolean }> => {
-      const fileName = createS3FileName(cxId, patientId, doc.docUniqueId);
-
-      const { exists } = await s3Utils.getFileInfoFromS3(fileName, s3BucketName);
-
-      return {
-        docId: doc.docUniqueId,
-        exists,
-      };
-    })
-  );
-
-  const successfulDocs = docIdWithExist.flatMap(ref =>
-    ref.status === "fulfilled" && ref.value ? ref.value : []
-  );
-
-  const existentialDocRefs = documents.reduce(
-    (acc: ExistentialDocRefs, curr) => {
-      for (const succDoc of successfulDocs) {
-        if (succDoc.docId === curr.docUniqueId) {
-          if (succDoc.exists) {
-            acc.existingDocRefs = [...acc.existingDocRefs, curr];
-          } else {
-            acc.nonExistingDocRefs = [...acc.nonExistingDocRefs, curr];
-          }
-        }
+// Since we have most of the document contents when doing the document query,
+// we will store this in s3 and then upsert the reference to the s3 object in FHIR
+// when doing the doc retrieval
+async function storeInitDocRefInFHIR(
+  docRefs: DocumentWithMetriportId[],
+  cxId: string,
+  patientId: string
+) {
+  await Promise.allSettled(
+    docRefs.map(async docRef => {
+      try {
+        const docId = docRef.metriportId ?? "";
+
+        const FHIRDocRef = cqToFHIR(docId, docRef, patientId);
+
+        await upsertDocumentToFHIRServer(cxId, FHIRDocRef);
+      } catch (error) {
+        console.log(`Failed to store initial doc ref in FHIR: ${errorToString(error)}`);
+        capture.error(error, {
```

Comment:
Could send message instead of error so we can more easily relate Sentry alerts to the issue at hand